src/openrct2/interface/Screenshot.cpp

```diff
@@ -368,8 +372,7 @@ void screenshot_giant()
         }
         if (viewport.width >= 64000 || viewport.height >= 32000)
         {
-            context_show_error(STR_SCREENSHOT_FAILED, STR_NONE, {});
-            return;
+            throw std::exception();
```

Comment:
I don't see any other `throw std::exception();` in the codebase.
Could this if block be changed into a `Guard::Assert`?

Example:
`Guard::Assert(viewport.width < 64000 && viewport.height < 32000, "Screenshot too large")`

Reply:
This causes a crash to desktop if Guard::Assert is used

Reply:
`std::exception` should not be thrown by itself, always use a class that derives from it. In this case `std::domain_error` seems appropriate, although I think this should be handled differently anyway. These magic numbers just happen to work because it matches the max map size, instead you should fall back to the actual map's edges if the selection covers it all.

Reply:
Changed this to std::domain_error. That being said, this exception should never occur at all as of now. Right now it is only exists just in case so the game does not outright crash if that very unlikely case. Could remove this exception if this never occurs at all.